### ISSUE TITLE
Modifying the viewport

### ISSUE THREAD

**Issue by quin** [Jpadilla1/react-native-ios-charts#28]
is `setVisibleXRangeMaximum` supported yet?
https://github.com/PhilJay/MPAndroidChart/wiki/Modifying-the-Viewport

**Comment by Jpadilla1** [comment]
Hi! Currently only `setViewPortOffsets` is supported, but shouldn't be that hard to include `setVisibleXRangeMaximum`. I'll see what I can do today to add support.

**Comment by Jpadilla1** [comment]
You can take a look at my previous commit when I added support for `setViewPortOffsets` if you want to add support and submit a PR yourself. https://github.com/Jpadilla1/react-native-ios-charts/commit/471f9a042053bbfc24ae8ee48cee3912835c3d89

**Comment by quin** [comment]
Sure

**Comment by quin** [comment]
when trying to run ChartsExplorer:
<img width="374" alt="screen shot 2016-05-20 at 2 23 18 pm" src="https://cloud.githubusercontent.com/assets/7978950/15442503/73af3614-1e96-11e6-8cb5-c2b0322ec6ad.png">

😱

**Comment by Jpadilla1** [comment]
<img width="487" alt="screen shot 2016-05-21 at 12 32 32 am" src="https://cloud.githubusercontent.com/assets/3780146/15446341/96dec906-1eeb-11e6-84d4-dccad4c49b51.png">

Strange, worked fine for me

**Comment by Jpadilla1** [comment]
Hey @brah been at this issue for a while now and it doesn't seem as straight forward as I thought. The visibleXRange will be affected after the chart is drawn so we would need to call the function in componentDidMount. Still trying to find a good way to make this work with objective-c so the method can be called from the ref of the component.

**Comment by Jpadilla1** [comment]
http://stackoverflow.com/questions/37365885/react-native-swift-get-associated-uiview-in-manager-with-reacttag

**Comment by Jpadilla1** [comment]
<img width="487" alt="screen shot 2016-05-23 at 10 09 17 am" src="https://cloud.githubusercontent.com/assets/3780146/15473537/7ea330a4-20ce-11e6-9e5b-edfd6e52ade3.png">

Got it working :)

**Comment by quin** [comment]
@Jpadilla1 awesome! what was your solution?

**Comment by Jpadilla1** [comment]
First I added these imports to the Bridging-Header.h file

``` Objective-C
#import "RCTBridge.h"
#import "RCTViewManager.h"
#import "RCTUIManager.h"
#import "UIView+React.h"
```

Then I export a external method in the specific chart objective-c .m file.

``` Objective-C
RCT_EXTERN_METHOD(setVisibleXRangeMaximum:(nonnull NSNumber *)reactTag value:(CGFloat *)v);
```

Next I implemented the function in Swift. 

``` Swift
@objc func setVisibleXRangeMaximum(reactTag: NSNumber, value: CGFloat) {
    self.bridge.uiManager.addUIBlock { (uiManager: RCTUIManager!, viewRegistry:[NSNumber : UIView]!) in
      let view: RNBarChart = viewRegistry[reactTag] as! RNBarChart;
      view.setVisibleXRangeMaximum(value);
    }
  }
```

Added this to the JavaScript component for BarChart

``` JS
constructor(props) {
  super(props);
  this.setVisibleXRangeMaximum = this.setVisibleXRangeMaximum.bind(this);
}
setVisibleXRangeMaximum(value) {
  RNBarChartManager.setVisibleXRangeMaximum(findNodeHandle(this), value);
}
```

Finally, called the function on componentDidMount on the Bar.js file in the ChartsExplorer example.

``` JS
componentDidMount() {
  this.refs.chart.setVisibleXRangeMaximum(2);
}
```

**Comment by Jpadilla1** [comment]
Added support for that function for now in master until I find a better way to generalize this for every chart that supports it. Test it out and see if it works for you!

**Comment by Jpadilla1** [comment]
Included in `0.2.5` release.
